tests/PidTest.cpp

```diff
@@ -0,0 +1,331 @@
+/**
+ * @file PidTest.cpp
+ * @brief PidGainクラス, Pidクラスをテストする
+ * @author miyahara046
+ */
+
+#include "Pid.h"
+#include "gtest/gtest.h"
+namespace etrobocon2025_test {
+
+  // PidGainの正のゲインがそのまま格納されるかをテスト
+  TEST(PidGainTest, PidGainPlus)
+  {
+    double kp = 0.1;
+    double ki = 0.03;
+    double kd = 0.612;
+    double expected_kp = 0.1;
+    double expected_ki = 0.03;
+    double expected_kd = 0.612;
+    PidGain actualPidGain(kp, ki, kd);
+    EXPECT_DOUBLE_EQ(expected_kp, actualPidGain.kp);
+    EXPECT_DOUBLE_EQ(expected_ki, actualPidGain.ki);
+    EXPECT_DOUBLE_EQ(expected_kd, actualPidGain.kd);
+  }
+  // PidGainの0のゲインがそのまま格納されるかをテスト
+  TEST(PidGainTest, PidGainZero)
+  {
+    double kp = 0.0;
+    double ki = 0.0;
+    double kd = 0.0;
+    double expected_kp = 0.0;
+    double expected_ki = 0.0;
+    double expected_kd = 0.0;
+    PidGain actualPidGain(kp, ki, kd);
+    EXPECT_DOUBLE_EQ(expected_kp, actualPidGain.kp);
+    EXPECT_DOUBLE_EQ(expected_ki, actualPidGain.ki);
+    EXPECT_DOUBLE_EQ(expected_kd, actualPidGain.kd);
+  }
+
+  // PidGainの負のゲインが0に補正されるかをテスト
+  TEST(PidGainTest, PidGainMinus)
+  {
+    double kp = -0.5;
+    double ki = -0.2;
+    double kd = -0.3;
+    PidGain actualPidGain(kp, ki, kd);
+    double expected = 0.0;
+    EXPECT_DOUBLE_EQ(expected, actualPidGain.kp);
+    EXPECT_DOUBLE_EQ(expected, actualPidGain.ki);
+    EXPECT_DOUBLE_EQ(expected, actualPidGain.kd);
+  }
+
+  // 基本的なPID計算の結果が正しいかをテスト
+  TEST(PidTest, CalculatePid)
+  {
+    Pid actualPid(0.6, 0.02, 0.03, 70, 100.0, -100.0);
+    double currentValue = 20;
```

Comment:
```suggestion
    double currentValue = 20.0;
```
ここらへんも double だから 20.0 がいいかな。